frontend/ui/uimanager.lua

```diff
@@ -714,7 +707,9 @@ function UIManager:handleInput()
 
     -- delegate input_event to handler
     if input_event then
-        self:_resetAutoSuspendTimer()
+        if input_event.handler ~= "onInputError" then
+            self:broadcastEvent(Event:new("InputEvent"))
```

Comment:
To the hook itself. It can be removed once the Widget is closed. But I doubt this assumption can be easily broken by some developers. It looks bad to have an API which forces developers to do too many things.
A good API:
  - Do this.
A bad API:
  - You need to do this first.
  - Do this in condition A.
  - Do this in condition B.
  - ...
  - If there are several more objects, you need to do this.
  - ...
  - Finally you need to do this once the object can be released.